### PR TITLE
Feat/timedmetadata ingest

### PR DESCRIPTION
**Pull request by andreasgangso**
The main juice is in
asset/timedmetadata.go
asset/timedmetadata.images.go
backend/cmd/jobs/tests/timedmetadata_test.go

I put the test in own package since its closer to an integration test than a unit test. It was much easier and more realistic to run it for real. So its useful for local testing while developing but cant be used in a ci/cd test suite.

### CODE REVIEW

**Review by KillerX — COMMENTED** [review]
I think it makes sense

File: queries/contributions.sql [hunk 1]
```diff
@@ -48,4 +48,8 @@ WHERE
     OR (roles.roles_earlyaccess && @roles::varchar[])
   )
 ORDER BY
-  m.published_at DESC;
\ No newline at end of file
+  m.published_at DESC;
+
+-- name: InsertContribution :exec
+INSERT INTO "public"."contributions" (person_id, "type", mediaitem_id, timedmetadata_id)
+VALUES (@person_id::uuid, @type, sqlc.narg('mediaitem_id')::uuid, sqlc.narg('timedmetadata_id')::uuid);
```
[the remark this answers is not in the document]
Comment (andreasgangso) [reply queries/contributions.sql:55]: not same thing. ::uuid is not nullable. `narg` makes it nullable

[the remark this answers is not in the document]
Comment (andreasgangso) [reply queries/contributions.sql:55]: https://docs.sqlc.dev/en/stable/howto/named_parameters.html#nullable-parameters
"there is no nullable equivalent of the @-syntax"

File: backend/asset/timedmetadata.images.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/ansel1/merry/v2"
+	"github.com/bcc-code/bcc-media-platform/backend/files"
+	"github.com/bcc-code/mediabank-bridge/log"
+	"golang.org/x/sync/errgroup"
+)
+
+// ingestImagesFromS3 downloads images from S3 and uploads them to the platform
+func ingestImagesFromS3(ctx context.Context, imagePaths []string, services externalServices, config config) (map[string]string, error) {
+	s3client := services.GetS3Client()
+	tempDir, err := os.MkdirTemp(config.GetTempDir(), "timedmetadata")
```
Comment (KillerX) [inline backend/asset/timedmetadata.images.go:18]: Is there a specific reason we are not using `os.TempDir()` but are instead introducing a whole new config var?

File: backend/asset/timedmetadata.images.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			}
+
+			imageId, err := uploadFileToPlatform(ctx, services, localPath)
+			if err != nil {
+				return merry.Wrap(err)
+			}
+			mu.Lock()
+			imageIDs[i] = *imageId
+			mu.Unlock()
+			return nil
```
Comment (KillerX) [inline backend/asset/timedmetadata.images.go:48]: Is there a direct need to download the file to disk? It seems like the stream could be passed directly from the "dowload" to the "upload" functions.

**Review by KillerX — APPROVED** [review]

File: backend/asset/timedmetadata.images.go
Comment (KillerX) [inline backend/asset/timedmetadata.images.go:1]: Please don't introduce a 3rd file naming :D `timedmetadata_images.go` is the most go-like

**Review by KillerX — APPROVED** [review]
Some comments for potential improvements, but lgtm in general. Up to you if you want to do something about it.

File: backend/asset/timedmetadata.images.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			})
+			if err != nil {
+				return merry.Wrap(err)
+			}
+
+			imageId, err := uploadFileToPlatform(ctx, services, localPath)
+			if err != nil {
+				return merry.Wrap(err)
+			}
+			mu.Lock()
```
Comment (KillerX) [inline backend/asset/timedmetadata.images.go:45]: Why do we need a lock here? I it not safe if we know that we are writing into different keys?

File: backend/asset/timedmetadata.images.go (hunk 3 above)
Comment (andreasgangso) [reply backend/asset/timedmetadata.images.go]: nope. changed to do that

**Pull request merged**